tests/json_infra/conftest.py

```diff
@@ -106,6 +109,13 @@ def pytest_addoption(parser: Parser) -> None:
         ),
     )
 
+    parser.addoption(
```

Comment:
Some of these options might end up contradicting each other. For example, we might have `--fork=Frontier` and the tests file might have an entry called `tests/json_infra/fixtures/latest_fork_tests/fixtures/state_tests/berlin/eip2930_access_list/test_account_storage_warm_cold_state.json::tests/berlin/eip2930_access_list/test_acl.py::test_account_storage_warm_cold_state[fork_Shanghai-state_test-account_warm_True-storage_key_warm_True]::Shanghai::0` which is from `Shanghai`.

In this set-up, the Shanghai tests will not run. Is this what is intended? Alternatively, we could just make it invalid to pair the `--tests-file` option with the others.